request/multiple_kernels.rst

```diff
@@ -0,0 +1,183 @@
+
+Adding a new Kernel to the Application
+--------------------------------------
+
+Creating a new kernel in a Symfony application is a three-step process:
+
+1. Create a new front controller to load the new kernel;
+2. Create the new kernel class;
+3. Define the configuration loaded by the new kernel.
+
+The following example shows how to create a new kernel for the API of a given
+Symfony application.
+
+Step 1) Create a new Front Controller
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Instead of creating the new front controller from scratch, it's recommended to
+duplicate the existing ones. For example, create ``web/api_dev.php`` from
+``web/app_dev.php`` and ``web/api.php`` from ``web/app.php``.
+
+Then, update the code of the new front controllers to instantiate the new kernel
+class instead of the usual ``AppKernel`` class::
+
+    // web/api.php
+    // ...
+    $kernel = new ApiKernel('prod', false);
+    // ...
+
+    // web/api_dev.php
+    // ...
+    $kernel = new ApiKernel('dev', true);
+    // ...
+
+Step 2) Create the new Kernel Class
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Now you need to define the ``ApiKernel`` class used by the new front controller.
+The recommendation again is to duplicate the existing ``app/AppKernel.php`` file
+and make the needed changes.
+
+In this example, the changes of the new ``ApiKernel`` would be to load less
+bundles than ``AppKernel`` and to change the location of the cache, logs and
+config files to not mess with the regular application::
+
+    // app/ApiKernel.php
+    <?php
+
+    use Symfony\Component\HttpKernel\Kernel;
+    use Symfony\Component\Config\Loader\LoaderInterface;
+
+    class ApiKernel extends Kernel
+    {
+        public function registerBundles()
+        {
+            // load only the bundles strictly needed for the API...
+        }
+
+        public function getCacheDir()
+        {
+            return dirname(__DIR__).'/var/cache/api/'.$this->getEnvironment();
+        }
+
+        public function getLogDir()
+        {
+            return dirname(__DIR__).'/var/logs/api';
+        }
+
+        public function registerContainerConfiguration(LoaderInterface $loader)
+        {
+            $loader->load($this->getRootDir().'/config/api/config_'.$this->getEnvironment().'.yml');
+        }
+    }
+
+Step 3) Define the Kernel Configuration
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Finally, define the configuration used by the application when it executes the
+new API kernel. According to the previous code, this config must be defined in
+the ``app/config/api/`` directory.
+
+The new configuration can be created from scratch when you load just a few
+bundles, because it will be very simple. Otherwise, duplicate the existing
+config files or better, import them and override the needed options:
+
+.. code-block:: yaml
+
+    # app/config/api/config_dev.yml
+    imports:
+        - { resource: ../config_dev.yml }
+
+    # override option values ...
+
```

Comment:
Another option is to *share* some configuration files, especially if both kernels use some of the same bundles that you want configured in the same ways. I'm haven't done this enough to settle on the best organization for this... but we could at least mention it. Perhaps a `app/config/shared.yml` file where we show just a little bit of config and mention how you could import this from the `config.yml` file of both kernels.